### DISCUSSION TITLE
Tus-uploady - Set file specs in Upload Metadata

### DISCUSSION THREAD

**Discussion by Tekpakma** [rpldy/react-uploady#441] [Q&A] [answered]
I am trying to provide file-specific Meta Data like type and name in the Upload-Metadata header. 
Tbh not really sure how to achieve that on demand modifications of params.

Can someone point me into the right direction?

**Comment by yoavniran** [comment, marked as answer]
Hey @Tekpakma 
You'd do that just like with other types of uploads. Using the [useRequestPreSend](https://react-uploady.org/docs/api/hooks/useRequestPreSend/) hook

simple example:

```javascript
 useRequestPreSend(({ items }) => {
        const name = items[0].file.name;

        return {
            options: {
                params: {
                    "md-fileName": name,
                }
            }
        }
    });
```

You can check out this [guide](https://react-uploady.org/docs/guides/DynamicParameters/) for more info as well.

Let me know if you have more questions.

**Reply by Tekpakma** [reply]
Cool worked like a charm! Also, nicely documented, sadly missed it ...

**Reply by NiharR27** [reply]
Hi @yoavniran - how do we go with multiple file upload (and of different types)? How do we pass the different mime types / file names? Thanks 🙏

**Reply by yoavniran** [reply]
hi @NiharR27 
Unless grouping is enabled and for TUS it cannot be. The `items` array should only have one upload at a time. Therefore, you can add headers/params and they will be sent to the server for that one specific upload.
